build-tools/automation/yaml-templates/apk-instrumentation.yaml

```diff
@@ -6,31 +6,49 @@ parameters:
   extraBuildArgs: ""
   testResultsFormat: NUnit
   packageType: Apk
-  artifactName: ""
+  artifactSource: ""
   artifactFolder: ""
+  useDotNetPreview: false
   condition: succeeded()
 
 steps:
-- task: MSBuild@1
-  displayName: run ${{ parameters.testName }}
-  inputs:
-    solution: ${{ parameters.project }}
-    configuration: ${{ parameters.configuration }}
-    msbuildArguments: >-
-      /restore
-      /t:AcquireAndroidTarget,SignAndroidPackage,DeployTest${{ parameters.packageType }}s,CheckAndRecordApkSizes,RunTestApks,UndeployTestApks,RenameApkTestCases,ReportComponentFailures
-      /bl:$(System.DefaultWorkingDirectory)/bin/Test${{ parameters.configuration }}/run-${{ parameters.testName }}.binlog
-      ${{ parameters.extraBuildArgs }}
-  condition: ${{ parameters.condition }}
-  continueOnError: true
+- ${{ if ne(parameters.useDotNetPreview, true) }}:
+  - task: MSBuild@1
+    displayName: run ${{ parameters.testName }}
+    inputs:
+      solution: ${{ parameters.project }}
+      configuration: ${{ parameters.configuration }}
+      msbuildArguments: >-
+        /restore
+        /t:AcquireAndroidTarget,SignAndroidPackage,DeployTest${{ parameters.packageType }}s,CheckAndRecordApkSizes,RunTestApks,UndeployTestApks,RenameApkTestCases,ReportComponentFailures
+        /bl:$(System.DefaultWorkingDirectory)/bin/Test${{ parameters.configuration }}/run-${{ parameters.testName }}.binlog
+        ${{ parameters.extraBuildArgs }}
+    condition: ${{ parameters.condition }}
+    continueOnError: true
+
+- ${{ if eq(parameters.useDotNetPreview, true) }}:
+  - powershell: |
+      $dotnetPath = "$env:HOME/Library/Android/dotnet/dotnet"
+      if ([Environment]::OSVersion.Platform -eq "Unix") {
+          if (-not (Test-Path $dotnetPath)) {
+              $dotnetPath = "$env:HOME/android-toolchain/dotnet/dotnet"
+          }
+      } else {
+          $dotnetPath = "$env:USERPROFILE\android-toolchain\dotnet\dotnet.exe"
+      }
```

Comment:
Is it worth trying to put some of this code in a `.yaml` template, we have something similar here:

https://github.com/xamarin/xamarin-android/blob/681887ebdbd192ce7ce1cd02221d4939599ba762/build-tools/automation/yaml-templates/run-nunit-tests.yaml#L17-L20

Something that sets a `$(DotNet.Tool.Path)` variable?

Reply:
I added a template for running our dotnet preview to improve reuse, with comparable parameters to https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/dotnet-core-cli?view=azure-devops